baybe/searchspace/discrete.py

```diff
@@ -635,26 +635,22 @@ def get_candidates(
     def transform(
         self,
         data: pd.DataFrame,
+        allow_missing: bool = False,
     ) -> pd.DataFrame:
-        """Transform parameters from experimental to computational representation.
-
-        Continuous parameters and additional columns are ignored.
-
-        Args:
-            data: The data to be transformed. Must contain all specified parameters, can
-                contain more columns.
-
-        Returns:
-            A dataframe with the parameters in computational representation.
-        """
+        """See :func:`baybe.searchspace.core.SearchSpace.transform`."""
         # If the transformed values are not required, return an empty dataframe
         if self.empty_encoding or len(data) < 1:
             comp_rep = pd.DataFrame(index=data.index)
             return comp_rep
 
         # Transform the parameters
         dfs = []
-        for param in self.parameters:
+        parameters = (
+            [p for p in self.parameters if p.name in data]
+            if allow_missing
+            else self.parameters
+        )
+        for param in parameters:
             comp_df = param.transform_rep_exp2comp(data[param.name])
```

Comment:
since youre at it once do you also want to change the name of the parameter transformation names away from `transform-rep_exp2comp` ? Its one of the oldest bit of code I remember writing

Reply:
can do. I assume you'd also just prefer a simple `transform` in this case? So if I do it, I should probably add to #289 , right? 

And then let me ask upfront: deprecation or not? (In this case, deprecation would be a simple name alias)